org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Grid.java

```diff
@@ -0,0 +1,114 @@
+package org.code.neighborhood;
+
+import java.util.ArrayList;
+import java.io.File;  // Import the File class
+import java.io.FileInputStream; // Import the Scanner class to read text files
+import java.lang.System;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class Grid {
+    private static Grid instance = null;
+    private GridSquare[][] grid;
+    private int width;
+    private int height;
+    private int numPainters;
+
+    public static Grid getInstance() {
+        if (instance == null) {
+            File file = new File("grid.txt");
+            FileInputStream fis;
+            try {
+                fis = new FileInputStream(file);
+                byte[] data = new byte[(int) file.length()];
+                fis.read(data);
+                fis.close();
+//                JSONParser parser = new JSONParser();
+                instance = new Grid(new String(data, "UTF-8"));
+            } catch (Exception e) {
+                System.out.println("\nTry adding a grid.txt file.\n");
+                e.printStackTrace();
+                System.exit(1);
+            }
+        }
+        return instance;
+    }
+
+    private Grid(String gridDescription) {
+        String[] lines = gridDescription.split("\n");
```

Comment:
It might be nice to separate out how a grid is created from what a grid does.  I'm imagining a constructor that takes just GridSquare[][].  There would then be a separate static constructor (static method that returns a Grid instance) that creates a Grid given a filename/string.  If you wanted further separation, that code could live in a separate class called `GridFactory`.  (Among other things, this means the Grid class would not need to import any file related dependencies.)

If we do all that, then unit testing the functionality in Grid becomes straightforward -- you create a Grid manually using the constructor and hand-crafting a GridSquare[][].  Then you can separately unit test the static constructor.